ed_sensor_integration/src/kinect/fitter.cpp

```diff
@@ -363,11 +351,22 @@ std::unique_ptr<OptimalFit> Fitter::findOptimum(const EstimationInputData& input
             if (!evaluateCandidate(input_data, candidate))
                 continue;
 
+            // Calculate error
+            double error = computeFittingError(candidate.test_ranges, input_data.sensor_ranges);
+
             // Update optimum
-            updateOptimum(input_data.sensor_ranges, candidate, *result);
+            if (error < current_optimum->getError())
+            {
+                current_optimum->update(candidate.pose, error);
+                // reject an optimum value found at the boundary of the search space as it is not a global maximum.
+                valid_optimum = !(i_beam==0 || i_beam==nr_data_points_-1);
+            }
```

Comment:
When the optimum is not valid, we don't have a any optimum at all.

Why not ignore the the invalid optimums and take the best valid optimum?

Reply:
We are testing whether the optimum is on the boundary. it is not a matter of a valid candidate but a valid result over the optimisation